### PR TITLE
New cloud account endpoints

### PR DESCRIPTION
**Pull request by calevans**

### CODE REVIEW

File: CloudAccount.md [hunk 1]
```diff
@@ -90,46 +104,214 @@ __Payload 1__
...
-| `cloud_id` | The Nexcess id for the cloud (data center) you want to spin up your new account within.  See '[Listing Clouds](Clouds.md)' to get a list of available `cloud_id` values. | YES |
-| `install_app` | tells the system whether or not to actually install the application you requested or to only prepare the server environment for you to install the application later. | NO |
+|`app_id`|the system id for the application to be install See [Applications](Applications.md) to get a list of the available `app_id` values. | YES |
+| `package_id` | the system id for the server package to be spun up. See '[Listing Packages](Packages.md)' to get a list of available `package_id` values. | YES |
+| `cloud_id` | The system id for the cloud (data center) to spin up the new account in.  See '[Listing Clouds](Clouds.md)' to get a list of available `cloud_id` values. | YES |
+| `install_app` | "ON" or "OFF".<br >Tells the system whether or not to actually install the application requested or to only prepare the server environment.  | NO |
```
Comment (robertology) [inline CloudAccount.md]: this is a boolean input. our api accepts multiple boolean values (`true`, `1`, `yes`, `on` i believe. Adrian can confirm). should we indicated `boolean` in these cases and have info on what that means in a central location?

in general for the docs, a `type` column would be good for these

File: CloudAccount.md [hunk 2]
```diff
@@ -284,43 +467,38 @@ This will return to you a very large payload that will give you details about bo
...
-- **stable** The final state of the cloud account is stable. This is your signal that everything truly is 200 OK.
+| State | Description |
+| :--- | :--- |
+| **creating** | The initial state. The cloud account will stay in this state until the initial creation of the cloud account has completed. |
+| **destroying** | The state of a machine that has been required to be destroyed and approved. It is in this state until the destroy process is complete. After this is complete the API will return 404 on all future calls to `/cloud-host/cloud_id` |
+| **failure** | If for some reason the creating process did not complete successfully, the cloud account will be assigned the state of failure. |
+| **installing** | Once creating is complete, if `install_app` was set to `on`, the application be installed. The cloud account will be put in the **installing** state. It will remain in this state until the application is successfully installed. |
+| **creating** | The final state of the cloud account is stable. This is the signal that everything truly is 200 OK. |
```
Comment (robertology) [inline CloudAccount.md]: `stable`

File: CloudAccount.md [hunk 3]
```diff
@@ -386,37 +566,265 @@ __Payload 5__
 }
 ```
 
-The payload returned in most cases describes the backup that will be created. Backups are an out-of-bandwidth process in almost all cases. In a very few cases, the backup will complete before the API returns. In these cases - very small sites or no files/DB ot be backed up - you will see that `complete` comes back as **true**.
+The payload returned in most cases describes the backup that will be created. Backups are an out-of-bandwidth process in almost all cases. In a very few cases, the backup will complete before the API returns. In these cases - very small sites or no files/DB to be backed up - `complete` will be returned as **true**.
 
-In most cases, `complete` will come back as **false**. To check the status of your backup:
+In most cases, `complete` will come back as **false**. To check the status of a backup:
 
 - Save off the file name
```
Comment (robertology) [inline CloudAccount.md]: `save off` is oddly worded to me

File: CloudAccount.md (hunk 1 above)
Comment (calevans) [reply CloudAccount.md]: If it's boolean, I would suggest that we only return boolean. (I've got a ticket open on this) For users who don't RTFM, "on" and "off" may be offputting, esp those who are not PHP developers. (Although I am a DPHP dev and I find it a bit odd) I will update the docs.

Comment (calevans) [reply CloudAccount.md]: Idea about the Type column noted. I will add.

File: CloudAccount.md [hunk 4]
```diff
@@ -284,43 +467,38 @@ This will return to you a very large payload that will give you details about bo
...
-- `copy_account` is a boolean flag. If set to true then the environment of the production cloud account will be copied into the development environment. This is an optional parameter. If not present, false is assumed.
-- `scrub_account` is a boolean flag. If set to true then data in the database will be anonymized. This is an optional parameter. If not present, false is assumed.
+__Parameters__
 
-__Example 5__
+| Name | Description | Type | Required |
+| :--- | :--- | :---: | :---: |
+| `domain` | Has to be a sub-domain of the production cloud account's domain. | String | YES |
+| `package_id` | The service_id associated with the parent cloud account. | Integer | YES |
+| `ref_type` | `development` | YES |
```
Comment (robertology) [inline CloudAccount.md]: missing type

File: CloudAccount.md [hunk 5]
```diff
@@ -386,37 +566,265 @@ __Payload 5__
...
+
+### Resize
+
+A cloud account can be resized both up and down. If the cloud account is a shared account, the resizing will take place in the background and a switch will be made once its state is "stable". This results in almost zero downtime. If the cloud account is a dedicated account then it will be taken off-line and resized.
+
+__Parameters__
+
+| Name | Description | Type | Required |
+| :--- | :--- | :---: | :---: |
+| `_action` | `resize` | YES |
```
Comment (robertology) [inline CloudAccount.md]: missing type

**Pull request merged**